src/main/java/mops/businesslogic/DirectoryServiceImpl.java

```diff
@@ -108,10 +108,13 @@ public UserPermission getPermissionsOfUser(Account account, long dirId) throws M
      * {@inheritDoc}
      */
     @Override
-    @SuppressWarnings("PMD.LawOfDemeter")
-    public Directory getOrCreateRootFolder(Account account, long groupId) throws MopsException {
-        // TODO: get the root folder if it exists
-        roleService.checkIfRole(account, groupId, ADMIN);
+    @SuppressWarnings({ "PMD.LawOfDemeter", "PMD.OnlyOneReturn" })
+    public Directory getOrCreateRootFolder(long groupId) throws MopsException {
+        Optional<Directory> optionalDirectory = directoryRepository.getRootFolder(groupId);
+        if (optionalDirectory.isPresent()) {
+            return optionalDirectory.get();
+        }
+
         Set<String> roleNames = permissionService.fetchRolesInGroup(groupId);
         DirectoryPermissions rootPermissions = createDefaultPermissions(roleNames);
```

Comment:
can you check here if roleNames is NOT empty and fail with an exception otherwise?

Reply:
Where exactly check? Before line 114, or between 116 and 118? Or before 119?

Reply:
where it makes sense. everyone may get the root directory, but it may only be created if the group exists